oshi-core/src/main/java/oshi/util/ParseUtil.java

```diff
@@ -1120,4 +1120,21 @@ public static String parseUtAddrV6toIP(int[] utAddrV6) {
             return Constants.UNKNOWN;
         }
     }
+
+    /**
+     * Parses a string of hex digits to long value.
+     * @param hexString A sequence of hex digits
+     * @param defaultValue default value to return if parsefails
+     * @return The corresponding long value
+     */
+    public static long hexStringToLong(String hexString, long defaultValue) {
```

Comment:
This looks great.   As the failed "codecov" prompts say, you need to write a testcase  (one passing one failing) in `ParseUtilTest`.

Reply:
I am unable to convert memory addres to long using this method as this throws `NumberFormatException`, however using `BigInteger`, i can get the long value, but this long value is overflowing. I still suggest that we should have the `address` as `String`.

Reply:
Ah, the problem is that it is an unsigned long on the C side which is bigger than max long.  We can still use a negative value.   I think there is actually already another parsing method in ParseUtil does character by character parsing and bit-shifting to do this.  

I’ll rewrite that parse method, you can just have it return 0 for now and move in to Solaris.

Reply:
The existing method `ParseUtil.byteArrayToLong()` will do what you need.  Just convert the String to a byte array using `ParseUtil.hexStringToByteArray()`.

Although it would be nicer to parse the hex string directly to the `long` instead of using the array, using `hexStringToByteArray()` as a starting point and adding/bitshifting bytes instead of storing in an array.

Reply:
I tried adding both the functions, and the result is same `
byte[] byteArray = hexStringToByteArray("ffffffff830f53a0"); 
 long result = byteArrayToLong(byteArray, 8);` **and** `new BigInteger("ffffffff830f53a0", 16).longValue()`both return the same value, which makes me think, why are we not using `String` for memory address. I think then i can easily picture a `memory address` just by looking at its value. Returning a long value is fine but when i look at it, it doesn't make any sense to make.

Reply:
Yes, the string value is more user friendly but we can leave that to the user to format how they want with %x on String.format().  The underlying data is a numeric value.  If not long, I’d consider a Pointer (which wraps a long as it’s peer value).

The value you get from big integer long value is the correct one that we want to return.

Reply:
Also the long only takes 8 bytes to store.  A string would take 8 bytes for the pointer to the memory location of the String object (a long!) plus all the overhead of an object plus the bytes for the String (2bytes per hex character turning 16 characters into 32 bytes).  :)

Reply:
You bring up a good point on readability and we should use the %x format in the toString().

Reply:
done.